File: REST-API-solution/REST_lib/ProductUpdate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace REST_lib
{
    public class ProductUpdate
    {

        //********************************************************************************
        // Data Properties
        //********************************************************************************
        public int ProductId { get; set; }
        public int QuantityToBeRemoved { get; set; }
        //********************************************************************************


        //********************************************************************************
        // Constructors
        //********************************************************************************
        public ProductUpdate()
        {
            ProductId = 0;
            QuantityToBeRemoved = 0;
        }

        public ProductUpdate(int ProductId, int QuantityToBeRemoved)
        {
            this.ProductId = ProductId;
            this.QuantityToBeRemoved = QuantityToBeRemoved;
        }
        //********************************************************************************



        //********************************************************************************
        // Override Functions
        #region override_functions_region 
        //********************************************************************************
        public override bool Equals(object obj)
        {
            if (obj == null)
                return false;
            if (obj.GetType() != this.GetType())
                return false;

            ProductUpdate temp = (ProductUpdate)obj;
            return (this.ProductId == temp.ProductId && this.QuantityToBeRemoved == temp.QuantityToBeRemoved);
        }
        public override int GetHashCode()
        {
            unchecked
            {
                int hash = 17 + QuantityToBeRemoved + ProductId;
                // Suitable nullity checks etc, of course :)
                hash = hash * 23 + ProductId.GetHashCode();
                hash = hash * 23 + QuantityToBeRemoved.GetHashCode();
                hash = hash * 23 + this.ToString().GetHashCode();
                return hash;
            }
        }

        public override string ToString()
        {
            return "ID:\t" + ProductId + "\tQuantity:\t" + QuantityToBeRemoved;
        }
        #endregion
        //********************************************************************************


    }
}

File: Client-JavaFX/src/main/java/data/ProductUpdate.java
package data;

import static java.util.Objects.hash;

public class ProductUpdate {

    public int productId;
    public int quantityToBeRemoved;

    public ProductUpdate()
    {
        productId = 0;
        quantityToBeRemoved = 0;
    }

    public ProductUpdate(int productId, int quantityToBeRemoved)
    {
        this.productId = productId;
        this.quantityToBeRemoved = quantityToBeRemoved;
    }

    @Override
    public int hashCode() {

        return hash(super.hashCode(), this.productId,this.quantityToBeRemoved);
    }

    @Override
    public boolean equals(Object obj) {
        ProductUpdate temp = (ProductUpdate)obj;
        return this.productId == temp.productId && this.quantityToBeRemoved == temp.quantityToBeRemoved;
    }

    @Override
    public String toString() {
        return "Product ID:\t" + this.productId + "\tPurchaseQuantity:\t" + this.quantityToBeRemoved;
    }
}
